### DISCUSSION TITLE
Missing all badges

### DISCUSSION THREAD

**Discussion by codecat** [Chatterino/chatterino2#4456] [Q&A] [answered]
It seems like in all chats (even in my own channel) I am missing all user badges on chat messages. I have the badge visibility settings turned on, and I also have message highlighting for mods and VIP's which **does** work, so it feels like the visibility is maybe bugged in some way?

Is there any way I can troubleshoot this? It's been like this for a couple couple days now and I don't see any reports of this issue here on Github, so this might just be something on my end being weird.

**Comment by Felanbird** [comment, marked as answer]
Do you have a VPN on, specifically PiA? 

edit: or anything that would block "spam/tracking"?

**Reply by codecat** [reply]
No, I am not using a VPN, but I was using ProtonVPN for a little while before. Could that somehow have messed up the cache?

Edit: I have some blocklists on my DNS (PiHole) and have recently made some changes to my setup, but disabling it doesn't seem to make a difference. So maybe it is cached somewhere. 🤔

**Reply by Felanbird** [reply]
~~Unlikely, my first thought was the fact that Twitch's badge endpoint is blocked on the grounds of tracking (that endpoint specifically is not tracking you), by anti-spam/tracking services.~~

~~Since you knew to think it was a caching issue, have you cleared your cache?~~

edit: Ah yes it is your PiHole, check for badges.twitch.tv in your blocklists, it was in Blocklisten/easylist last I checked https://github.com/RPiList/specials/issues/851

**Reply by codecat** [reply]
Interesting, it seems you are right! I am having trouble with that domain. That issue is pretty old though, maybe my DNS had an old list installed? Or maybe it is an issue again?

Thank you either way, I am digging a little bit more into this! 👏

**Reply by Felanbird** [reply]
@codecat While ur looking into it, can you let me know if https://t.3v.fi/badges.html loads. I'd like to know if #4071 is the same issue, although they _were_ able to load the badges on the page, despite the fact that they likely shouldn't have been able to.

**Reply by codecat** [reply]
So it turns out that [The Block List Project](https://github.com/blocklistproject/Lists) has `badges.twitch.tv` in its `ads` list. I've opened a new issue on their Github here: https://github.com/blocklistproject/Lists/issues/950
